Dockerfile

```diff
@@ -15,4 +15,5 @@ COPY --from=0 /usr/local /usr/local
 WORKDIR /app
 COPY .bigqueryrc /root/
 COPY . .
+COPY target/sql/ sql/
```

Comment:
@jklukas re https://github.com/mozilla/bigquery-etl/pull/177#issuecomment-499982301

>> this should be merged at the same time as a PR to telemetry-airflow that updates https://github.com/mozilla/telemetry-airflow/blob/master/dags/utils/gcp.py#L400 to point to target/sql/ instead of sql/
>
> I had imagined we update the docker build script in this repo instead to copy the local target/sql tree into sql in the image. In that case, the docker image wouldn't need to change.

I think we should follow the example set by [mozilla-pipeline-schemas](https://github.com/mozilla-services/mozilla-pipeline-schemas#notes)

> ... the artifacts are generated/saved back into the repository ... The reason for this is twofold:
> 1. It lets us easily see and refer to complete schemas as they are actually used. This means that the schemas can be referenced directly in bugs and such, as well as being fetched directly from the repo for testing other schema consumers (test being important here, as any production use should be using the installable packages).
> 2. It gives us a changelog for each schema, rather than having to reason about changes to templated external pieces and when/how that impacted a given doctype's schema over time. This means that it should be easy to look back in time for the provenance of different parts of the schema for each doctype.

additionally:

- artifacts from this repo will include not only adding UDFs into queries, but eventually also other generated sql, such as queries for [clients_daily_scalar_aggregates](https://github.com/mozilla/bigquery-etl/pull/168/files#diff-008d245a3022a42ec2ebffe07d6f6eea) and [longitudinal](https://github.com/mozilla/bigquery-etl/blob/611bd30/sql/longitudinal.sql.py)
- if we rename the files in the container, either pytest will need to behave differently inside docker vs outside of it, or we won't be able to run pytest on the resulting container

Reply:
I still find the build workflow slightly confusing, so not sure if the new changes I made to the `Dockerfile` make sense and resolve this.

Reply:
> builds should be reproducible by running `docker build .` on a clean git checkout

This concern is addressed

> artifacts [should be] generated/saved back into the repository

> if we rename the files in the container ... we won't be able to run pytest on the resulting container

These concerns are what i'm talking about in this thread, and both should wait to be addressed until @jklukas and I come to an agreement about them

Reply:
I spoke with @relud about this in person, and we agreed that following the example of mozilla-pipeline-schemas would be the best way forward here, which involves a refactoring of directories. Instead of reading from `sql` to generate `target/sql`, we'll move the sql files to a new `templates` directory and have the generated files be placed in `sql/` (in m-p-s, the templates live in `templates/` and the generated files live in `schemas/`).

We'll expect that generated files are checked in to the repository.

Reply:
After we spoke on Friday, I remembered that m-p-s has a test that runs file generation and verifies that it matches the content checked into git. We should be able to copy this almost directly from m-p-s, and it will ensure that developers are correctly generating files before merging a PR:

https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/master/.circleci/config.yml#L25-L43

Reply:
Okay, interesting. I made the changes and updated the CircleCI config accordingly.